sorts/quicksort.ts

```diff
@@ -0,0 +1,51 @@
+/* 
+Implementation of in-place quick sort with stack.
+
+  complexity:
+    best time - O(N log N)
+    avg time - O(N log N)
+    worst time - O(N^2)
+    
+    memory - O(log N)
+    in-place - Yes
+    stable - No
+*/
+
+module.exports = function QuickSort(arr: number[]): number[] {
+  const stack: number[][] = [];
+  if (arr.length > 1) stack.push([0, arr.length - 1]);
+  while (stack.length) {
+    // left and right boundaries of unsorted subsequences
+    const [left, right] = stack.pop();
+    let l = left,
+      r = right,
+      m = Math.floor((l + r) / 2);
```

Comment:
I'd call it "pivot" not "middle".

Reply:
Good Idea

Reply:
No, this should be called middle, because we will determine the pivot later in algorithm